lib/repl.js

```diff
@@ -1209,31 +1201,22 @@ function complete(line, callback) {
     if (!subdir) {
       completionGroups.push(_builtinLibs);
     }
-
-    completionGroupsLoaded();
   } else if (fsAutoCompleteRE.test(line)) {
-    filter = '';
+    let baseName = '';
     let filePath = line.match(fsAutoCompleteRE)[1];
-    let fileList;
+    let fileList = gracefulReaddir(filePath, { withFileTypes: true });
 
-    try {
-      fileList = fs.readdirSync(filePath, { withFileTypes: true });
-      completionGroups.push(fileList.map((dirent) => dirent.name));
-      completeOn = '';
-    } catch {
-      try {
-        const baseName = path.basename(filePath);
-        filePath = path.dirname(filePath);
-        fileList = fs.readdirSync(filePath, { withFileTypes: true });
-        const filteredValue = fileList.filter((d) =>
-          d.name.startsWith(baseName))
-          .map((d) => d.name);
-        completionGroups.push(filteredValue);
-        completeOn = baseName;
-      } catch {}
+    if (!fileList) {
```

Comment:
Since you already did 95% of the work I'd extract the recovery to a function

Reply:
I moved the whole code that handles fs calls. That way it's clear what that code part does. Does that solve the comment?